packages/api/src/router/target.ts

```diff
@@ -357,6 +363,36 @@ export const targetRouter = createTRPCRouter({
           total,
         }));
       }),
+
+    overlappingTargets: protectedProcedure
+      .meta({
+        authorizationCheck: ({ canUser, input }) =>
+          canUser
+            .perform(Permission.TargetList)
+            .on({ type: "workspace", id: input.workspaceId }),
+      })
+      .input(overlappingTargetsInput)
+      .query(async ({ ctx, input }) => {
+        const { workspaceId, filterA, filterB } = input;
+        const workspaceIdCheck = eq(schema.target.workspaceId, workspaceId);
+        const combinedConditions: SQL<unknown>[] = [
+          workspaceIdCheck,
+          ...ensureArray(schema.targetMatchesMetadata(ctx.db, filterA)),
+          ...ensureArray(schema.targetMatchesMetadata(ctx.db, filterB)),
+        ];
+        const overlappingItems = await targetQuery(
+          ctx.db,
+          combinedConditions,
+        ).then((t) =>
+          t.map((a) => ({
+            ...a.target,
+            provider: a.targetProvider,
+            metadata: a.targetMetadata,
+          })),
+        );
+
+        return overlappingItems;
+      }),
```

Comment:
why do we need an endpoint for this if we can use our existing list targets query

Reply:
Our current list query only has a `limit: z.number().default(500),` 

I wanted something that would work with more, but then I would have to add pagination logic to the list and still bump the limit because the pagination still works within that limit. If we bump the limit and did pagination I would need to do a lot more work on the fronted.

Reply:
Done